[MOD-84][MOD-126] Empty action lists

aaxelb opened the pull request:
Handle empty action lists, and do a little cleanup. I ended up fixing a few things I thought I broke but in fact were already broken.

- upgrade to Ember 2.15
- add `loading-indicator` component, use it instead of all those silly divs
- don't use ember-data-has-many-query to query preprints on the moderation list page
    - it breaks when the first query returns 0 results
    - instead I added a simpler but largely equivalent `query` function that sends a request manually and pushes the results into the store
    - also fixes MOD-126!

<img width="970" alt="screen shot 2017-10-06 at 4 27 40 pm" src="https://user-images.githubusercontent.com/6776190/31297260-62691352-aab3-11e7-8209-b36024292c7c.png">
<img width="982" alt="screen shot 2017-10-06 at 4 32 37 pm" src="https://user-images.githubusercontent.com/6776190/31297461-0b8b567a-aab4-11e7-9284-878fb147665b.png">

https://openscience.atlassian.net/browse/MOD-84
https://openscience.atlassian.net/browse/MOD-126

aaxelb committed:
Update preprint status banner to handle no actions

aaxelb committed:
Try squashing some bugs

aaxelb committed:
Don't use ember-data-has-many-query

aaxelb committed:
yarn

aaxelb committed:
Fix list loading state

aaxelb committed:
Some cleanup

laurenbarker commented on app/components/loading-indicator/template.hbs:
```diff
@@ -0,0 +1,5 @@
+<div class="ball-pulse ball-dark {{classes}}" aria-label="Loading data">
```
`ball-dark` should probably be a passed in class because we don't always want the dark version (e.g. `contributor-list`).

laurenbarker commented on app/components/preprint-status-banner/template.hbs:
```diff
@@ -10,7 +10,15 @@
         <div class="recent-activity">
-            <a href={{creatorProfile}}>{{creatorName}}</a> {{t recentActivityLanguage}} {{moment-format submission.dateLastTransitioned "MMMM DD, YYYY"}}
+            {{#if loadingActions}}
+                {{loading-indicator classes='text-center m-b-lg p-v-md'}}
```
The extra padding isn't needed when loading:
![screen shot 2017-10-09 at 11 12 03 am](https://user-images.githubusercontent.com/7131985/31345094-b53686ce-ace2-11e7-9aa9-ef3ac39b860f.png)
![screen shot 2017-10-09 at 11 12 38 am](https://user-images.githubusercontent.com/7131985/31345117-ca8bfa22-ace2-11e7-975a-e8d5dec9ce91.png)

laurenbarker reviewed (changes requested):
Just a few tiny things 💯

aaxelb committed:
Fix transitions from detail to moderation list

aaxelb committed:
Respond to review

aaxelb commented:
@laurenbarker Ready for another look!

laurenbarker commented:
Clicking on a row before it's fully loaded results in contributors never being loaded:
![no-contributors](https://user-images.githubusercontent.com/7131985/31353330-27a59476-ad00-11e7-9182-d5f6d67b6b05.gif)

laurenbarker commented:
Also weird things are happening when transitioning from the dashboard (at least for me):
![odd-transitions](https://user-images.githubusercontent.com/7131985/31354719-689e434c-ad05-11e7-9172-b05a71ac7294.gif)

laurenbarker commented on app/routes/preprints/provider/moderation.js:
```diff
@@ -4,6 +4,38 @@ import Ember from 'ember';
+            }, reject);
+        } else {
+            const message = `Could not find a link for '${propertyName}' relationship`;
+            console.error(message, 'on model:\n', model);
```
Travis doesn't like this

laurenbarker reviewed (changes requested):
Seems like some weird behavior was introduced with the transition aborted fix.

aaxelb committed:
Responding further to review

aaxelb commented:
@laurenbarker I was able to replicate that first bug (clicking on a row while it's still loading causing contributors not to load) on staging3, seems worth a separate ticket?

The weird transitions to/from the dashboard are because I added `templates/loading.hbs`. I removed it because it does behave weirdly enough that it's better to just be slow. We should maybe think about making more route-specific loading templates, if the model hooks are slow enough on staging/prod.

laurenbarker commented:
Ah, okay. Yeah I'll add a ticket for route specific loading pages and the contributors bug.
